tools/upload_to_server.py

```diff
@@ -0,0 +1,232 @@
+            " Run 'build' and 'make-installer' first."
+        )
+
+    with open(metadata_path, "r") as stream:
+        metadata = json.load(stream)
+
+    installer_path = metadata.get("installer_path")
+    if not installer_path or not os.path.exists(installer_path):
+        raise click.BadParameter(
+            "Installer is not available. Run 'make-installer' first."
+        )
+    metadata["filename"] = os.path.basename(installer_path)
+    return InstallerInfo(**metadata)
+
+
+def create_connection(
+    server: str, api_key: str, username: str, password: str
+) -> ayon_api.ServerAPI:
+    api = ayon_api.ServerAPI(server)
+    if not api.is_server_available:
+        raise click.BadParameter("Server is not available")
+
+    if not api_key and not (username and password):
+        raise click.BadParameter(
+            "You must provide API key, or username and password",
+            param_hint="api-key, username & password"
+        )
+
+    if api_key:
+        api.set_token(api_key)
+        if api.has_valid_token:
+            return api
+
+        if not username or not password:
+            raise click.BadParameter(
+                "API key is not valid."
+                " Provide valid API key, or username and password.",
+                param_hint="api-key"
+            )
+
+    api.login(username, password)
+    if api.has_valid_token:
+        return api
+    raise click.BadParameter(
+        "Invalid credentials."
+        " Provide valid API key, or username and password.",
+        param_hint="username & password"
+    )
+
+
+def create_installer(
+    api: ayon_api.ServerAPI, installer_info: InstallerInfo, force:bool
+):
+    server_installers: list[dict[str, Any]] = (
+        api.get_installers()["installers"])
+
+    matched_installer = None
+    for server_installer in server_installers:
+        platform_name: Union[str, None] = server_installer.get("platform")
+        version: Union[str, None] = server_installer.get("version")
+        if (
+            platform_name == installer_info.platform
+            and version == installer_info.version
+        ):
+            matched_installer = InstallerInfo(
+                version=server_installer.get("version"),
+                platform=server_installer.get("platform"),
+                filename=server_installer.get("filename"),
+                installer_path=installer_info.installer_path,
+                python_version=server_installer.get("pythonVersion"),
+                checksum=server_installer.get("checksum"),
+                checksum_algorithm=server_installer.get("checksumAlgorithm"),
+                size=server_installer.get("size"),
+                python_modules=server_installer.get("pythonModules"),
+                runtime_python_modules=(
+                    server_installer.get("runtimePythonModules")),
+            )
+            break
+
+    if matched_installer is not None:
+        if matched_installer == installer_info:
```

Comment:
This could maybe be done at line `114` ?

Reply:
I guess that was before the change?